src/connect/connect.js

```diff
@@ -29,16 +30,24 @@ function match(arg, factories) {
   return undefined
 }
 
+function strictEqual(a, b) { return a === b }
+
 export function buildConnectOptions(
   mapStateToProps,
   mapDispatchToProps,
   mergeProps,
   {
-    mapStateToPropsFactories = defaultMapStateToPropsFactories,
-    mapDispatchToPropsFactories = defaultMapDispatchToPropsFactories,
-    mergePropsFactories = defaultMergePropsFactories,
-    selectorFactory = defaultSelectorFactory,
     pure = true,
+    areStatesEqual = strictEqual,
+    areOwnPropsEqual = shallowEqual,
+    areStatePropsEqual = shallowEqual,
+    areMergedPropsEqual = shallowEqual,
+    factories: {
```

Comment:
I don't really agree with this. I'd rather keep the options map flat, even though it's somewhat large. It keeps things relatively symmetrical when dealing with them.